src/solver/closure.go

```diff
@@ -19,8 +23,50 @@ func IsSupportedFunction(name string) bool {
 	return ok
 }
 
-// Struct containing information about the function that called the closure.
+// ClosureCallerInfo containing information about the function that called the closure.
 type ClosureCallerInfo struct {
 	Name     string          // caller function name
 	ArgTypes []meta.TypesMap // types for each arg for call caller function
 }
+
+func GetClosureName(fun *ir.ClosureExpr, curFunction, curFile string) string {
+	pos := ir.GetPosition(fun)
+	if curFunction != "" {
+		curFunction = "," + curFunction
+	}
+	name := `\Closure(` + curFile + curFunction + "):" + fmt.Sprint(pos.StartLine)
+	return name
+}
+
+func GetClosure(name ir.Node, sc *meta.Scope, cs *meta.ClassParseState) (meta.FuncInfo, bool) {
+	if !cs.Info.IsIndexingComplete() {
+		return meta.FuncInfo{}, false
+	}
+
+	nmf, ok := name.(*ir.SimpleVar)
+	if !ok {
+		return meta.FuncInfo{}, false
+	}
+
+	var fi meta.FuncInfo
+	var found bool
+	var sv = &ir.SimpleVar{Name: nmf.Name}
+
+	tp := ExprTypeLocal(sc, cs, sv)
+	tp.Iterate(func(typ string) {
```

Comment:
Maybe you want to use `Find` instead of `Iterate` here?

Reply:
Yep. Done.